src/textual/_wait.py

```diff
@@ -0,0 +1,27 @@
+from asyncio import sleep
+from time import process_time, monotonic
+
+SLEEP_GRANULARITY: float = 1 / 50
+SLEEP_IDLE: float = SLEEP_GRANULARITY / 2.0
+
+
+async def wait_for_idle(
+    min_sleep: float = SLEEP_GRANULARITY, max_sleep: float = 1
+) -> None:
+    """Wait until the cpu isn't working very hard.
+
+    Args:
+        min_sleep: Minimum time to wait. Defaults to 1/50.
+        max_sleep: Maximum time to wait. Defaults to 1.
+    """
+    start_time = monotonic()
+
+    while True:
+        cpu_time = process_time()
+        await sleep(SLEEP_GRANULARITY)
+        cpu_elapsed = process_time() - cpu_time
+        elapsed_time = monotonic() - start_time
+        if elapsed_time >= max_sleep:
+            break
+        if elapsed_time > min_sleep and cpu_elapsed < SLEEP_IDLE:
+            break
```

Comment:
I have no clue of what's going on.

Your PR says short sleeps are problematic but then there's a nice contrast between `elapsed_time > min_sleep` and `cpu_elapsed < SLEEP_IDLE`.

This feels like a function that needs a couple more comments or a more comprehensive docstring.